### PR TITLE
KEYCLOAK-17209 Add example for configuring ldap

### PR DESCRIPTION
**Pull request by soleblaze**
## JIRA ID
https://issues.redhat.com/browse/KEYCLOAK-17209

## Additional Information

This adds an example for how to configure ldap while storing the bind credential in the plaintext vault via a mounted secret.

## Checklist:
- [ ] Automated Tests
- [ ] [Documentation](https://github.com/keycloak/keycloak-documentation) changes if necessary
- [ ] (If required) Discussed on [Keycloak DEV](https://groups.google.com/forum/#!forum/keycloak-dev)

### CODE REVIEW

**Comment by soleblaze** [comment]
This example uses secret mounting, which is currently waiting to be merged via #314. Once that PR is merged then this one should be good to go.

File: deploy/examples/keycloak/keycloak-with-vault.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: keycloak-vault
+type: Opaque
+stringData:
+  test-realm_bindCredential: "test-password"
```
Comment (kfox1111) [inline deploy/examples/keycloak/keycloak-with-vault.yaml:8]: Should this be ldap-realm_bindCredential?

**Comment by kfox1111** [comment]
This also fixes: https://issues.redhat.com/projects/KEYCLOAK/issues/KEYCLOAK-17091 :)

File: deploy/examples/realm/realm_with_ldap.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+          ldap.attribute: mail
+          read.only: 'true'
+          user.model.attribute: email
+      - name: full name
+        federationProviderDisplayName: ldap
+        federationMapperType: full-name-ldap-mapper
+        config:
+          ldap.full.name.attribute: cn
+          read.only: 'true'
+          write.only: 'false'
```
Comment (kfox1111) [inline deploy/examples/realm/realm_with_ldap.yaml:80]: So I'm understanding this right, and maybe would be good to add to the example, would adding this mapper to pull in ldap groups work here?
```
- name: group
  federationProviderDisplayName: ldap
  federationMapperType: group-ldap-mapper
  config:
    membership.attribute.type: DN
    group.name.ldap.attribute: cn
    membership.user.ldap.attribute: sAMAccountName
    preserve.group.inheritance: 'true'
    groups.dn: DC=example,DC=com
    mode: READ_ONLY
    user.roles.retrieve.strategy: GET_GROUPS_FROM_USER_MEMBEROF_ATTRIBUTE
    groups.ldap.filter: "(cn=EXAMPLE*)" #Only pull in groups that start with prefix EXAMPLE
    membership.ldap.attribute: "true"
    memberof.ldap.attribute: memberOf
    group.object.classes: group
    drop.non.existing.groups.during.sync: "true"
```

File: deploy/examples/keycloak/keycloak-with-vault.yaml (hunk 1 above)
Comment (soleblaze) [reply deploy/examples/keycloak/keycloak-with-vault.yaml:8]: It should. Thanks for pointing that out. I fixed it.

File: deploy/examples/realm/realm_with_ldap.yaml (hunk 2 above)
Comment (soleblaze) [reply deploy/examples/realm/realm_with_ldap.yaml:80]: That's right. I'll add yours as an example. I'm using this one for talking to my openldap server:

```
            - name: "groups"
              federationMapperType: group-ldap-mapper
              config:
                drop.non.existing.groups.during.sync: 'false'
                group.name.ldap.attribute: cn
                group.object.classes: posixGroup
                groups.dn: "ou=Groups,dc=example,dc=com"
                ignore.missing.groups: 'false'
                memberof.ldap.attribute: "memberOf"
                membership.attribute.type: "UID"
                membership.ldap.attribute: "memberUid"
                membership.user.ldap.attribute: "uid"
                mode: "READ_ONLY"
                preserve.group.inheritance: 'false'
                user.roles.retrieve.strategy: "LOAD_GROUPS_BY_MEMBER_ATTRIBUTE"
```

**Comment by kfox1111** [comment]
Do you have a KeycloakClient CR with groups injected into the id token yet? I've only been able to do it via manually configuring Keycloak to do it so far.

**Comment by soleblaze** [comment]
I haven't got that far yet. I'll take a look when I start setting up clients.

**Comment by kfox1111** [comment]
Not sure it will help or not, as I haven't found a 1:1 mapping between realm export and Kubernetes CR's, but when I added the mapper to the client I get this in the protocolMappers section:
```
        {
          "id": "c385b290-15a3-481f-9f3e-d6aecc56fea0",
          "name": "groups",
          "protocol": "openid-connect",
          "protocolMapper": "oidc-usermodel-realm-role-mapper",
          "consentRequired": false,
          "config": {
            "multivalued": "true",
            "user.attribute": "foo",
            "id.token.claim": "true",
            "access.token.claim": "true",
            "claim.name": "groups",
            "jsonType.label": "String"
          }
        },
```

**Comment by soleblaze** [comment]
@kfox1111 I opened a JIRa here: https://issues.redhat.com/browse/KEYCLOAK-17236 for additional documentation. I'll answer your question there so we don't clutter up this PR with our continued conversation.

**Review by slaskawi — APPROVED** [review]

**Comment by kfox1111** [comment]
Do you have an example of a KeycloakClient that maps in the groups?

**Comment by kfox1111** [comment]
Ping

**Comment by vmuzikar** [comment]
@kfox1111 I'm not sure I understand the question. Clients don't map groups.

This is not the right place to ask questions. Please use either our [forum](https://keycloak.discourse.group/) or [keycloak-user ML](https://groups.google.com/g/keycloak-user). Thank you.

**Comment by kfox1111** [comment]
For a keycloak client, I often will add a groups mapper to add the users groups to the id token. I need to figure out how to do the same thing with the KeycloakClient object.  I'll post on the forum. Thanks for the pointer.

**Comment by kfox1111** [comment]
https://keycloak.discourse.group/t/keycloak-operator-kubeclient-groups-mapper-example/10625

**Pull request merged**